tests/main/hidden-snap-dir/task.yaml

```diff
@@ -8,13 +8,96 @@ environment:
     NAME: test-snapd-tools
 
 prepare: |
-    snap pack "$TESTSLIB"/snaps/"$NAME"
+    snap pack "$TESTSLIB/snaps/$NAME"
     "$TESTSTOOLS"/snaps-state install-local "$NAME"
 
 restore: |
     snap unset system experimental.hidden-snap-folder
 
 execute: |
+    # core22 snap isn't available for x86
+    if os.query is-pc-i386; then
+      exit 0
+    fi
+
+    # Checks that the env vars are as expected after the migration.
+    check_env() {
+      echo "Check the env vars were migrated"
+
+      local CHECK_EXPOSED_HOME="no"
+      if [ "$1" = "--with-exposed-home" ]; then
+        CHECK_EXPOSED_HOME="yes"
+        shift
+      fi
+      local REV="$1"
+
+      snapEnv=$("$NAME".env)
+      echo "$snapEnv" | MATCH "SNAP_USER_DATA=$HOME/\.snap/data/$NAME/$REV"
+      echo "$snapEnv" | MATCH "SNAP_USER_COMMON=$HOME/\.snap/data/$NAME/common"
+
+      if [ "$CHECK_EXPOSED_HOME" = "yes" ]; then
+        echo "Checking core22 migration related env vars"
+        echo "$snapEnv" | MATCH "HOME=$HOME/Snap"
+        echo "$snapEnv" | MATCH "SNAP_USER_HOME=$HOME/Snap"
+        echo "$snapEnv" | MATCH "XDG_DATA_HOME=$HOME/\.snap/data/$NAME/$1/data"
+        echo "$snapEnv" | MATCH "XDG_CACHE_HOME=$HOME/\.snap/data/$NAME/$1/cache"
+        echo "$snapEnv" | MATCH "XDG_CONFIG_HOME=$HOME/\.snap/data/$NAME/$1/config"
+      else
+        echo "$snapEnv" | MATCH "HOME=$HOME/\.snap/data/$NAME/$1"
```

Comment:
nitpick: you forgot to replace `$1` with `$REV` here :-)

Reply:
Thanks for catching that. Fixed it now